src/main/java/com/amazon/opendistroforelasticsearch/search/asynchronous/service/AsynchronousSearchService.java

```diff
@@ -422,10 +422,11 @@ private void cancelAndFreeActiveAndPersistedContext(AsynchronousSearchActiveCont
      * @param asynchronousSearchContext the active asynchronous search context
      * @return boolean indicating if the state machine moved the state to CLOSED
      */
-    // TODO make this package private
```

Comment:
I think `initStateMachine` and others in `AsyncSearchService` should also be package private

Reply:
`initStateMachine` is private
Most methods in `AsyncSearchService` are invoked from transport layer and hence are required to be public

Reply:
I'm sorry I meant `getStateMachine`